Add guides for submitting PRs when experiencing the impostor phenomenon #773 (#807)
microsoft/code-with-engineering-playbook: docs/code-reviews/inclusion-in-code-review.md

We as CSE, may come across situations in which code reviews are not ideal and often we are observing non inclusive code review behaviors. Its important to be aware of the fact that culture and communication style of a particular geography also influences how people interact over pull requests.
In such cases, assuming positive intent of the author and reviewer is a good start to start analyzing quality of code reviews.

## Dealing with the Impostor Phenomenon

Impostor phenomenon is a psychological pattern in which an individual doubts their skills, talents, or accomplishments and has a persistent internalized fear of being exposed as a "fraud" - [wikipedia](https://en.wikipedia.org/wiki/Impostor_phenomenon).

> Someone experiencing impostor phenomenon may find submitting a code review particularly stressful. It is important to realise that everybody can have meaningful contributions and to not let the perceived weaknesses prevent said contributions.

Reviewer: NIT: perhaps just: prevent contributions?

Author: Changed, thank you.

Some tips for overcoming the impostor phenomenon for authors:

> * Review the guidelines highlighted above and make sure your code change adhere the them.

Reviewer: NIT: to them

Author: Changed, thank you.

> * Ask for help from a colleague - pair program with an experienced colleague that you can learn from.

Reviewer: I would also add not only pair program but also do the code review in pairing and not only by giving comments.
It is more personal and a discussion can be made about the code.

Author: I added a new line for reviewers.

Some tips for overcoming the impostor phenomenon for reviewers:

> * Anyone can have valuable insights. A fresh new pair of eyes are always welcome. Study the review until you have clearly understood it, check the corner cases and look for ways to improve it.

Reviewer: I would split this one with * as well:
* Anyone can have valuable insights. 
* A fresh new pair of eyes are always welcome. 
* Study the review until you have clearly understood it, check the corner cases and look for ways to improve it.

Author: Looks better indeed.

We as CSE, may come across situations in which code reviews are not ideal and often we are observing non inclusive code review behaviors. Its important to be aware of the fact that culture and communication style of a particular geography also influences how people interact over pull requests.
In such cases, assuming positive intent of the author and reviewer is a good start to start analyzing quality of code reviews.

## Dealing with the Impostor Phenomenon

> Impostor phenomenon is a psychological pattern in which an individual doubts their skills, talents, or accomplishments and has a persistent internalized fear of being exposed as a "fraud" - [wikipedia](https://en.wikipedia.org/wiki/Impostor_phenomenon).

Reviewer 2: ```suggestion
Impostor phenomenon is a psychological pattern in which an individual doubts their skills, talents, or accomplishments and has a persistent internalized fear of being exposed as a "fraud" - [Wikipedia](https://en.wikipedia.org/wiki/Impostor_phenomenon).
```

> Someone experiencing impostor phenomenon may find submitting a code review particularly stressful. It is important to realise that everybody can have meaningful contributions and to not let the perceived weaknesses prevent contributions.

Reviewer 2: ```suggestion
Someone experiencing impostor phenomenon may find submitting code for a review particularly stressful. It is important to realize that everybody can have meaningful contributions and not to let the perceived weaknesses prevent contributions.
```

Reviewer 2: Since the Wikipedia quote in the beginning of the section is in American English, I'd have it like so for the rest of the section too. Hence, the "nit".

Some tips for overcoming the impostor phenomenon for authors:

* Review the guidelines highlighted above and make sure your code change adhere to them.
* Ask for help from a colleague - pair program with an experienced colleague that you can learn from.

Some tips for overcoming the impostor phenomenon for reviewers:

* Anyone can have valuable insights.
* A fresh new pair of eyes are always welcome.
* Study the review until you have clearly understood it, check the corner cases and look for ways to improve it.
* If something is not clear, a simple specific question should be asked.
* If you have learnt something, you can always compliment the author.

> * If possible, pair with someone for code reviewing so that you establish a personal connection and have a more complex discussion about the code.

Reviewer 2: ```suggestion
* If possible, pair with someone to review the code so that you can establish a personal connection and have a more profound discussion about the code.
```
